src/sys/socket/addr.rs

```diff
@@ -2327,7 +2327,10 @@ mod tests {
 
         #[test]
         fn from_sockaddr_un_named() {
+            #[cfg(not(target_os = "haiku"))]
```

Comment:
This socket never actually gets constructed.  The test just exercises sockaddr type conversions.  So why special case it for Haiku?